server/data/data-jmap-postgres/src/main/java/org/apache/james/jmap/postgres/change/PostgresMailboxChangeModule.java

```diff
@@ -0,0 +1,73 @@
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.postgres.change;
+
+import static org.apache.james.jmap.postgres.change.PostgresMailboxChangeModule.PostgresMailboxChangeTable.INDEX;
+import static org.apache.james.jmap.postgres.change.PostgresMailboxChangeModule.PostgresMailboxChangeTable.TABLE;
+
+import java.time.OffsetDateTime;
+import java.util.UUID;
+
+import org.apache.james.backends.postgres.PostgresIndex;
+import org.apache.james.backends.postgres.PostgresModule;
+import org.apache.james.backends.postgres.PostgresTable;
+import org.jooq.Field;
+import org.jooq.Record;
+import org.jooq.Table;
+import org.jooq.impl.DSL;
+import org.jooq.impl.SQLDataType;
+
+public interface PostgresMailboxChangeModule {
+    interface PostgresMailboxChangeTable {
+        Table<Record> TABLE_NAME = DSL.table("mailbox_change");
+
+        Field<String> ACCOUNT_ID = DSL.field("account_id", SQLDataType.VARCHAR.notNull());
+        Field<UUID> STATE = DSL.field("state", SQLDataType.UUID.notNull());
+        Field<OffsetDateTime> DATE = DSL.field("date", SQLDataType.TIMESTAMPWITHTIMEZONE.notNull());
+        Field<Boolean> IS_SHARED = DSL.field("is_shared", SQLDataType.BOOLEAN.notNull());
+        Field<Boolean> IS_COUNT_CHANGE = DSL.field("is_count_change", SQLDataType.BOOLEAN.notNull());
+        Field<UUID[]> CREATED = DSL.field("created", SQLDataType.UUID.getArrayDataType().notNull());
+        Field<UUID[]> UPDATED = DSL.field("updated", SQLDataType.UUID.getArrayDataType().notNull());
+        Field<UUID[]> DESTROYED = DSL.field("destroyed", SQLDataType.UUID.getArrayDataType().notNull());
+
+        PostgresTable TABLE = PostgresTable.name(TABLE_NAME.getName())
+            .createTableStep(((dsl, tableName) -> dsl.createTableIfNotExists(tableName)
+                .column(ACCOUNT_ID)
+                .column(STATE)
+                .column(DATE)
+                .column(IS_SHARED)
+                .column(IS_COUNT_CHANGE)
+                .column(CREATED)
+                .column(UPDATED)
+                .column(DESTROYED)
+                .constraint(DSL.primaryKey(ACCOUNT_ID, STATE))))
+            .supportsRowLevelSecurity()
```

Comment:
Q: Should we disable RLS for email change and mailbox change?

( I am not sure but seeing DAOs keep being created during runtime makes me concerned about performance)
Likely the data in those two tables is not sensitive for domain isolation.

Reply:
Might agree with this. The info also is bound to disappear eventually with the TTL

Reply:
Likely can keep the RLS wherever we can: https://github.com/apache/james-project/pull/1941#discussion_r1465162089

Reply:
@chibenwa thoughts about this one? I know we said RLSs deeply linked to users so technically it should apply there, but as those entries  are for jmap notifications and not persistent (TTL), does it really make sense?

Reply:
To be honnest the accountId technically belong to a user linked to a domain. Which means that with this asumption in mind it should be doable to apply RLS (though we likely cannot get the domain from the AccountId).

So without an API change I think applying RLS would not be achieved.

As said above this is not that critical: just state changes... As such I'd agree not using RLS here if the setup is hard to achieve.

Reply:
> though we likely cannot get the domain from the AccountId

Hmm likely we can: `Username.of(accountId.getIdentifier()).getDomainPart()`
cf https://github.com/apache/james-project/blob/postgresql/server/data/data-jmap-postgres/src/main/java/org/apache/james/jmap/postgres/change/PostgresEmailChangeRepository.java#L113

Reply:
Then...


Let's go!